Sources/UI/ProfileImageView/ProfileImageView.swift

```diff
@@ -178,15 +204,109 @@ public extension ProfileImageView {
   }
 }
 
-// MARK: - Public Methods
-@available(iOS 14, *)
+// MARK: - Builder Pattern Methods
 public extension ProfileImageView {
+  func placeholder(_ placeholder: Image) -> Self {
+    properties.placeholder = placeholder
+    return self
+  }
+  
+  func backgroundType(_ backgroundType: Background) -> Self {
+    properties.backgroundType = backgroundType
+    return self
+  }
+  
+  func cornerRadius(_ cornerRadius: CornerRadiusType) -> Self {
+    properties.cornerRadius = cornerRadius
+    return self
+  }
+  
+  func contentMode(_ contentMode: ContentMode) -> Self {
+    properties.contentMode = contentMode
+    return self
+  }
+  
+  func borderColor(_ borderColor: Color) -> Self {
+    properties.borderColor = borderColor
+    return self
+  }
+  
+  func borderWidth(_ borderWidth: CGFloat) -> Self {
+    properties.borderWidth = borderWidth
+    return self
+  }
+  
+  func badging(_ badging: BadgingMode) -> Self {
+    properties.badging = badging
+    return self
+  }
+  
+  func url(_ url: URL) -> Self {
+    properties.url = url
+    return self
+  }
+  
+  func onBadgeTap(_ badgeTapped: @escaping (() -> Void)) -> Self {
+    self.properties.badgeTapped = badgeTapped
+    return self
+  }
+  
+  func onProfileTap(_ imageTapped: @escaping (() -> Void)) -> Self {
+    self.properties.imageTapped = imageTapped
+    return self
+  }
+}
+
+// MARK: - Access to properties from UIKit
+public extension ProfileImageView {
+  var placeholder: Image? {
+    get { properties.placeholder }
+    set { properties.placeholder = newValue}
+  }
+  
+  var backgroundType: Background {
+    get { properties.backgroundType }
+    set { properties.backgroundType = newValue }
+  }
+  
+  var cornerRadius: CornerRadiusType {
+    get { properties.cornerRadius }
+    set { properties.cornerRadius = newValue }
+  }
+  
+  var contentMode: ContentMode {
+    get { properties.contentMode }
+    set { properties.contentMode = newValue }
+  }
+  
+  var borderColor: Color {
+    get { properties.borderColor }
+    set { properties.borderColor = newValue }
+  }
+  
+  var borderWidth: CGFloat {
+    get { properties.borderWidth }
+    set { properties.borderWidth = newValue }
+  }
+  
+  var badging: BadgingMode {
+    get { properties.badging }
+    set { properties.badging = newValue }
+  }
+  
   func set(_ url: URL?) {
     properties.url = url
   }
+  
+  func addActionOnProfileTap(_ action: @escaping () -> Void) {
```

Comment:
Maybe we should rename this just to be "onProfileTap" like on the above function?